main.py

```diff
@@ -0,0 +1,60 @@
+from typing import List
+
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.engine import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/")
+async def root():
+    return {"message": "Hello World"}
+
+
+@app.get("/authors/", response_model=List[schemas.Author])
+def get_authors(skip: int = 0, limit: int = 10, db: Session = Depends(get_db)):
+    return crud.get_authors(db, skip=skip, limit=limit)
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorBaseCreate,
+    db: Session = Depends(get_db),
+):
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+
+    if db_author is not None:
+        raise HTTPException(status_code=400, detail="Author already exists")
+
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/books/", response_model=List[schemas.Book])
+def get_books(skip: int = 0, limit: int = 10, db: Session = Depends(get_db)):
+    return crud.get_books(db, skip=skip, limit=limit)
```

Comment:
Consider adding a query parameter to filter books by author ID. This will allow users to retrieve books associated with a specific author, which is currently missing from the implementation.